lib/mix/lib/mix/dep/loader.ex

```diff
@@ -301,17 +304,18 @@ defmodule Mix.Dep.Loader do
     |> elem(0)
   end
 
-  defp rebar_children(app, root_config, extra, manager) do
+  defp rebar_children(root_config, manager) do
     from = Path.absname("rebar.config")
     Mix.Rebar.recur(root_config, fn config ->
-      app
-      |> Mix.Rebar.deps(config, overrides(manager, extra))
-      |> Enum.map(fn dep -> %{to_dep(dep, from, manager) | extra: extra} end)
+      overrides = overrides(manager, config)
+      config
+      |> Mix.Rebar.deps()
+      |> Enum.map(fn dep -> %{to_dep(dep, from, manager) | extra: overrides} end)
```

Comment:
So the accumulation would not happen here but in `apply_overrides`. I.e. when we apply overrides on a given dependency, we need to make sure to also keep the overrides around for the next dependency. Can you please change `apply_overrides` accordingly and add some tests? Otherwise I can do that later when I am back to coding.